qemu/tests/qemu_guest_agent.py

```diff
@@ -1309,50 +1309,65 @@ def gagent_check_get_disks(self, test, params, env):
+                              "between guest and qga." % diskname)
             else:
-                if disk_info['dependencies'][0] != disk_info_guest['pkname']:
-                    test.fail("Disk %s dependencies is different "
+                if disk_info['name'] != disk_info_guest['kname']:
+                    test.fail("Disk %s name is different "
                               "between guest and qga." % diskname)
 
             error_context.context("Check partition", logging.info)
-            partition = False if disk_info_guest['type'] != "part" else True
-            if disk_info["partition"] != partition:
-                test.fail("Disk %s partition is different "
-                          "between guest and qga." % diskname)
-
-            if disk_info_guest['type'] == 'lvm':
-                error_context.context("Check alias", logging.info)
-                cmd_get_disk_alias = params["cmd_get_disk_alias"] % diskname
-                disk_alias = session.cmd_output(cmd_get_disk_alias).strip()
-                if disk_info['alias'] != disk_alias:
-                    test.fail("Disk %s alias is defferent "
+            if params.get("os_type") == "linux":
+                partition = False if disk_info_guest['type'] != "part" else True
+                if disk_info["partition"] != partition:
+                    test.fail("Disk %s partition is different "
                               "between guest and qga." % diskname)
+            else:
+                logging.info("From the disks in windows, "
+                             "the partition attributes are all false ")
+
+            if params.get("os_type") == "linux":
+                error_context.context("Check dependencies", logging.info)
+                if disk_info_guest['type'] == 'disk':
+                    dependencies = 'null'
+                    try:
+                        if disk_info['dependencies'][0] != dependencies:
+                            test.error("Disk %s dependencies "
+                                       "should be null." % diskname)
+                    except IndexError:
+                        logging.info("Disk %s dependencies is null", diskname)
+                    else:
+                        if disk_info['dependencies'][0] != disk_info_guest['pkname']:
+                            test.fail("Disk %s dependencies is different "
+                                      "between guest and qga." % diskname)
+
+                if disk_info_guest['type'] == 'lvm':
+                    error_context.context("Check alias", logging.info)
+                    cmd_get_disk_alias = params["cmd_get_disk_alias"] % diskname
+                    disk_alias = session.cmd_output(cmd_get_disk_alias).strip()
+                    if disk_info['alias'] != disk_alias:
+                        test.fail("Disk %s alias is defferent "
+                                  "between guest and qga." % diskname)
```

Comment:
The steps seems very different and so that you have too much `if...else` block, what do you think to have an independent function for windows?

Reply:
Yes, the same suggestion with qizhu